app/src/main/java/com/github/bgabriel998/softwaredevproject/points/UserPoint.java

```diff
@@ -18,7 +22,9 @@
  *
  * This class should be used as a observer that observes a GPSTracker.
  */
-public class UserPoint extends Point {
+public final class UserPoint extends Point {
+
+    private static final UserPoint single_instance = null; // singleton instance
```

Comment:
Should not be made final to assign a variable to single_instance
```suggestion
    private static UserPoint single_instance = null; // singleton instance
```